### ISSUE TITLE
Add an example to perform a QR factorization without storing Q

### ISSUE THREAD

**Issue by amontoison** [JuliaSmoothOptimizers/QRMumps.jl#88]
It's relevant when we want to solve `AA'x = b` <=> `R'Rx = b` with `A' = QR`.

**Comment by amontoison** [comment]
Yes, we should add new functions for that.

I also think we should incorporate a generic API (`cholesky`, `qr`, `lq`, `\`, `ldiv!`) for the `spmat` type.
I implemented this for `CuSparseMatrixCSR` in CUDSS.jl, which greatly simplifies tasks for the user: https://github.com/exanauts/CUDSS.jl/blob/main/src/generic.jl

We can introduce a keyword argument seminormal (or something similar) when creating a spmat to indicate that we want to solve systems with `AA'` or `A'A`.
The `\` operation will then handle the entire procedure for us (factorization + solve + iterative refinement).

**Comment by amontoison** [comment]
@abuttari
Alfredo, do you have an option to perform iterative refinement during the solve phase in `qr_mumps`?

**Comment by abuttari** [comment]
> @abuttari Alfredo, do you have an option to perform iterative refinement during the solve phase in `qr_mumps`?

nope, no iterative refinement in `qr_mumps` currently but shouldn't take too long to add.

**Comment by dpo** [comment]
We could simply do it in Julia. It’s not strictly necessary to add it to QRMumps itself, unless you want to.

**Comment by MaxenceGollier** [comment]
I added said documentation in PR #105, I don't use iterative refinement though, let me know if I should add it to the tutorial.
